### DISCUSSION TITLE
Problems in constructing two dimensional logical DSL

### DISCUSSION THREAD

**Discussion by Alsace-lee** [milvus-io/milvus#4611] [Q&A and General discussion]
I got a query logic like: ((release_year=2020 and duration=100) or duration>=120),which is a two dimensional logic.
then I build a DSL like this:

![image](https://user-images.githubusercontent.com/61733548/104427655-792be880-55be-11eb-93bd-f583046b6727.png)

However, the query result contains 0 vectors,  that is wired, because I can make sure that there has been such vectors in my milvus instance.

it seems like the logic between these _must_ nodes which is under _should_ node is AND rather than OR
Is there anything wrong with my DSL?

**Comment by miia12** [comment]
It seems like your dsl doesn't have syntactic errors. Could you try to remove one `must` restriction to see if there are search results?

**Reply by Alsace-lee** [reply]
In fact, neither of the two _must_ can be removed. Removing the first _must_ under should will cause the logic to be changed, and removing the second will cause the leaf node and Boolean node to be under should at the same time, which would throw a server side exception.

**Reply by miia12** [reply]
You should remove the upper must and should layers.

**Reply by miia12** [reply]
Are you using v0.11.0 or v0.11.1? There are a few bugs of dsl in v0.11.0.

**Reply by Alsace-lee** [reply]
> Are you using v0.11.0 or v0.11.1? There are a few bugs of dsl in v0.11.0.

we are using v0.11.0

**Reply by miia12** [reply]
If there still exist problems, I suggest you source compile 0.11.1, which we have fixed some bugs of dsl. 
However, currently we are doing a major refactor for 0.11.X, and support for 0.11.X will be limited to few pilot users. The source code will still be available from Github, you can still use it at your own risk.

> You should remove the upper must and should layers.

**Comment by miia12** [comment]
<img width="669" alt="截屏2021-01-14 上午11 04 34" src="https://user-images.githubusercontent.com/22544815/104539352-4b937d80-5658-11eb-8b0b-8fe9c52f9838.png">

`should` equals OR, and `must` equals AND.
